sale_disable_inventory_check/tests/test_disable_inventory_check.py

```diff
@@ -23,12 +41,23 @@ def test_disable_inventory_check(self):
             'state': 'draft',
         })
         # Sale product_1 with quantity is 15
-        order_line = self.env['sale.order.line'].new({
+        order_line1 = self.env['sale.order.line'].new({
             'order_id': order.id,
             'product_id': self.product_1.id,
             'product_uom': self.product_1.uom_id.id,
             'product_uom_qty': 15,
         })
-        res = order_line._onchange_product_id_check_availability()
-        self.assertEqual(res.get('warning', {}), {},
+        res1 = order_line1._onchange_product_id_check_availability()
+        self.assertEqual(res1.get('warning', {}), {},
                          "There should not be a warning!!!")
+        # Sale product_2 with quantity is 15
+        order_line2 = self.env['sale.order.line'].new({
+            'order_id': order.id,
+            'product_id': self.product_2.id,
+            'product_uom': self.product_2.uom_id.id,
+            'product_uom_qty': 15,
+        })
+        res2 = order_line2._onchange_product_id_check_availability()
+        self.assertEqual(
+            res2.get('warning', '').get('title', ''),
```

Comment:
You are defaulting to a string (`''`), but then you are calling `.get()` on it. Since this is a test, you can be more explicit about how exactly it fails:
```python
self.assertIn('warning', res2, "Should have returned a warning")
self.assertIsInstance(res2['warning'], dict, "Warning should be a dictionary")
self.assertIn('title', res2['warning'], "Warning should have a title")
self.assertEqual(res2['warning']['title'], 'Not enough inventory!', "There should be a warning!!!")
```